hamilton/include/hamilton/hybrid_planner.h

```diff
@@ -0,0 +1,133 @@
+    std::vector<geometry_msgs::Pose> waypoints_geom_msgs_;
+  }; 
+
+  /*  =============================== Hybrid Planner Class ===============================
+   * Provides a group of functions for planning a robot path using Moveit and  the Descartes Planning Library */
+
+  class HybridPlanner
+  {
+    public:
+      HybridPlanner();
+      virtual ~HybridPlanner();
+      // Initialize (descartes) robot model and planners
+      void init();
+
+    protected:
+      // Append unplanned free space/process path segments defined by "waypoints' to unplanned_trajectory_ as a TrajectorySegment object. 
+      // TODO extend to other types - joints, descartes types, etc.
+    	void appendFreeMotionTrajectorySegment(std::vector<geometry_msgs::Pose>& waypoints);
+    	void appendProcessPathTrajectorySegment(std::vector<geometry_msgs::Pose>& waypoints);
+
+      // Appends unplanned free space segment, defined by start and end point only as a TrajectorySegment object. 
+      void appendFreeMotionTrajectorySegment(geometry_msgs::Pose start_pose, geometry_msgs::Pose end_pose);
+
+      // Iterates through the unplanned_trajectory_vector, and calls MoveIt! and Descartes planners as required.
+      // Saves the planned, joint space trajectory in overall_robot_traj_, which can be send for execution to the controller.
+      // TODO Not Implemented: Makes a motion planning request if the TrajetorySegemnt.type_ == FREE_MOTION_START_END_ONLY
+    	void PlanHybrid();
+
+      //====== Helper functions =====//
+      /* Create an axial trajectory pt from a given tf transform */
+      descartes_core::TrajectoryPtPtr tfToAxialTrajectoryPt(const tf::Transform& nominal, double discretization);  
+
+      // Generates an completely defined (zero-tolerance) cartesian point from a pose
+      descartes_core::TrajectoryPtPtr makeCartesianPoint(const Eigen::Affine3d& pose);
+
+      // Generates a cartesian point with free rotation about the Z axis of the EEF frame
+      descartes_core::TrajectoryPtPtr makeTolerancedCartesianPoint(const Eigen::Affine3d& pose);
+
+      // Translates a descartes trajectory to a ROS joint trajectory
+      trajectory_msgs::JointTrajectory toROSJointTrajectory(const TrajectoryVec& trajectory, const std::vector<std::string>& joint_names, double time_delay);
+      //====== Helper functions =====//
+
+      //====== Member Variables =====//
+      HybridConfiguration config_;
+
+      // ROS Constructs: Components needed to successfully run a ros-node and perform other important ros-related tasks 
+      ros::NodeHandle node_handle_;                        /* Object used for creating and managing ros resources*/
+
+      // Descartes Constructs: Components accessing the semi-constrained path planning capabilities in the Descartes library */
+      descartes_core::RobotModelPtr descartes_model_;
+      descartes_planner::SparsePlanner descartes_planner_sparse_;      
+      descartes_planner::DensePlanner descartes_planner_dense_;
+
```

Comment:
For doing motion planning with moveit all that is needed is a service client that can send MotionPlanningRequest to  the move_group node.   Since there's no service server for the descartes planner as there is for moveit, then you'll have to store the Descartes planner(s) as a member of the HybridPlanner class.  Did that answer your inquiry?